packages/pipeline-editor/src/PipelineService.tsx

```diff
@@ -83,28 +92,49 @@ export class PipelineService {
       true
     );
 
-    const dialogTitle = 'Job submission to ' + runtimeName + ' succeeded';
-    const dialogBody = (
-      <p>
-        Check the status of your pipeline at{' '}
-        <a href={response['run_url']} target="_blank" rel="noopener noreferrer">
-          Run Details.
-        </a>
-        <br />
-        The results and outputs are in the {
-          response['object_storage_path']
-        }{' '}
-        working directory in{' '}
-        <a
-          href={response['object_storage_url']}
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          object storage
-        </a>
-        .
-      </p>
-    );
+    let dialogTitle;
+    let dialogBody;
+    if (response['run_url']) {
+      // pipeline executed remotely in a runtime of choice
+      dialogTitle = 'Job submission to ' + runtimeName + ' succeeded';
+      dialogBody = (
+        <p>
+          Check the status of your job at{' '}
+          <a
+            href={response['run_url']}
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            Run Details.
+          </a>
+          <br />
+          The results and outputs are in the {
+            response['object_storage_path']
+          }{' '}
+          working directory in{' '}
+          <a
+            href={response['object_storage_url']}
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            object storage
+          </a>
+          .
+        </p>
+      );
+    } else {
+      // pipeline executed in-place locally
+      dialogTitle = 'Job execution in-place succeeded';
```

Comment:
Doesn't this also get produced when the local job execution fails since you're not raising exceptions on failures?  How is the UI supposed to determine failure in these cases?

Reply:
No, this is only for success. Otherwise, the internal call show the error results with a different common dialog that displays the exception 

![image](https://user-images.githubusercontent.com/382917/90155514-45634300-dd40-11ea-92fa-507f42c01c06.png)